### ISSUE TITLE
`-invalid` suffix on post creation

### ISSUE THREAD

**Issue by frissdiegurke** [pitaj/nodebb-plugin-calendar#39]
If you already add an event to a new post creation (opposed to adding during edit), the `-invalid` suffix gets added.
I guess the cause of this is as following:
If you try to add an event during the post creation, NodeBB fires the [`filter:post.save`](https://github.com/NodeBB/NodeBB/blob/master/src/posts/create.js#L59) hook before [saving the post within the database](https://github.com/NodeBB/NodeBB/blob/master/src/posts/create.js#L62). This as I suppose results in privilege request returning 'not allowed' since the `pid` is not known in the system yet. You'll probably have to use `action:post.save` to check the privileges instead.

**Comment by pitaj** [comment]
There's another issue with `post.save` not passing modified values upstream (to sockets, other users, etc). Waiting on NodeBB/NodeBB#4875 for that fix.
